\begin{abstract}
\begin{description}
\item[Background]
Nucleotide and protein sequence feature annotation is essential to understand biology on the genome, transcriptome, and proteome level.
Unfortunately for using semantic web technologies to query biological annotation,
there was no standard that describes this potentially complex location information as subject-predicate-object triples using RDF/SPARQL.
\item[Description]
We developed a schema ontology to describe the positions of annotated features (FALDO)
that is usable for describing glycan binding sites, protein annotation and nucleotide features in sequence records.
Using the same data format to represent sequence positions independent of files allows us to integrate sequence data from multiple sources and data types.
The genome browser JBrowse is used to demonstrate accessing multiple SPARQL endpoints to display genomic feature annotation,
as well as protein annotation from UniProt mapped to genomic locations.
\item[Conclusions]
This standard ontology allows users to merge sequence annotation from multiple sources using federalised SPARQL queries against public endpoints and local private datasources.
At the same time the ontology is expressive enough to describe all known biological use cases accurately.
\end{description}
\end{abstract}
