\section*{Background}
Understanding how nucleotide and protein sequences in databases correspond to the real world is key to understanding biological processes.
Determined nucleotide sequences are stored in a large number of databases, such as GenBank/DDBJ/EMBL and vast numbers of new sequence records are generated every day since the advent of high throughput sequencing. Currently there is a vast number of ways of describing such sequences and their biological relation e.g. GenBank format, GFF3 and variants. All of these formats require the annotation to be stored in the same file as the sequence, making it difficult to merge information between files without copying all data.

While determining sequences has become vastly cheaper over time basic curation of annotation has not. This means that we still move most annotation from one sequence to another, e.g. gene annotation moves from one genome assembly to the next.

The proposed standard allows us to accurately describe the position of a feature on multiple sequences. For example a gene can start at position X on genome assembly A while the same gene is positioned on X' on genome assembly A+1. The proposed standard also deals with incomplete assemblies as well as protein sequence annotation such as found in UniProtKB.
