\section*{Background}
Describing regions of biological sequences is a vital part of genome and protein sequence
annotation, and beyond such as in describing modifications such methylation of nucleotide
sequences or glycosylation or proteins.
There are multiple different conventions for storing this kind of information in
plain text flat file formats such as GTF, GFF3, GenBank and EMBL,
and more structured domain specific XML formats such as the INSDC or UniProt XML.
As a result, file format interconversion is a common background task in current script-centric
bioinformatics pipelines, a necessary evil for combining tools supporting different formats or
format variables.

In the growing area of semantic web technologies, it was recognised at the BioHackathon 2012
meeting [REF Katayama et al 2013] that a failure to standardise how to describe positions
and regions on biological sequences would be an obstacle to the adoption of federalized
SPARQL/RDF queries which have the potential to enable cross-database queries and
analyses. Discussion and prototyping with representatives from major sequence databases
such as UniProt and DDBJ (part of the INSDC partnership with the NCBI and EMBL-Bank)
\textit{(TODO - and PDBj too?)} and assorted open source developers during this meeting
led to the development of the Feature Annotation Location Description Ontology (FALDO).

Understanding how nucleotide and protein sequences in databases correspond to the real world is key to understanding biological processes.
Determined nucleotide sequences are stored in a large number of databases,
such as GenBank/DDBJ/EMBL and vast numbers of new sequence records are generated every day since the advent of high throughput sequencing.
Currently there are a vast number of ways of describing such sequences and their biological relation e.g. GenBank format, GFF3 and variants.
All of these formats require the annotation to be stored in the same file as the sequence,
making it difficult to merge information between files without copying all data.

While determining sequences has become vastly cheaper over time basic curation of annotation has not.
This means that we still move most annotation from one sequence to another,
e.g. gene annotation moves from one genome assembly to the next.

The proposed standard allows us to accurately describe the position of a feature on multiple sequences.
For example a gene can start at position $X$ on genome assembly $A$,
while conceptually the same gene is positioned on $X'$ on genome assembly $A+1$.
The proposed standard also deals with incomplete assemblies as well as protein sequence annotation such as found in UniProtKB.

None of the above tools or formats are flexible enough to discuss all of genetics or proteomics. 

