\section*{Discussion}
When designing FALDO, a broad range of use cases were considered from
human genome annotation to glycan binding sites and protein domains on
amino acid sequences, with the goal of developing a scheme general enough
to describe regions of DNA, RNA and protein sequences.

Advantages and drawbacks of existing file formats were considered, including
line based column formats like BED and GTF/GFF which focus on exact
ranges on a given sequence, and the more complex locations supported
by the INSDC feature tables used by GenBank/EMBL/DDBJ.

The simplest non-stranded range location on a linear sequence requires
a start and end coordinate, but even here there are existing competing
conventions for describing open or closed end points using zero and
one based counting (for example BED versus GTF/GFF3/INSDC).

Similarly multiple schemes exist for describing strand specific locations,
with some formats describing features on the reverse strand implicitly
when the start coordinate is numerically higher than the end coordinate
\textit{(TODO - example of format which does this)}.

For a semantic description describing the strand explicitly is preferable.
FALDO chooses to add the strand information to the position. 
This is required to accurately describe for example the sticky ends
of an enzyme digestion cut site, as in the HindIII example (Figure~\ref{fig:HindIII}).

Unlike formats like GTF/GFF3, FALDO adopts the convention
that the start coordinate should be the biological start which 
may be numerically a higher value than the end coordinate.

FALDO has been designed to be open to extension for future
use \textit{TODO - expand on this with Jerven}.
