\section*{Discussion}
When designing FALDO, a broad range of use cases were considered from
human genome annotation to glycan binding sites and protein domains on
amino acid sequences, with the goal of developing a scheme general enough
to describe regions of DNA, RNA and protein sequences.

Advantages and drawbacks of existing file formats were considered, including
line based column formats like BED and GTF/GFF which focus on exact
ranges on a given sequence, and the more complex locations supported
by the INSDC feature tables used by GenBank/EMBL/DDBJ.

The simplest non-stranded range location on a linear sequence requires
a start and end coordinate, but even here there are existing competing
conventions for describing open or closed end points using zero and
one based counting (for example BED versus GTF/GFF3/INSDC).

Similarly multiple schemes exist for describing strand specific locations,
with some formats describing features on the reverse strand implicitly
when the start coordinate is numerically higher than the end coordinate
\textit{(TODO - example of format which does this)}.
In FALDO we always count from the start of the forward 5'-3' strand even on the reverse strand.
This encoding means there is no need to know the length of the sequence to compare positions on the different strands.

For a semantic description describing the strand explicitly is preferable.
FALDO chooses to add the strand information to the position. 
This is required to accurately describe for example the sticky ends
of an enzyme digestion cut site, as in the HindIII example (Figure~\ref{fig:HindIII}).

Unlike formats like GTF/GFF3, FALDO adopts the convention
that the start coordinate should be the biological start which 
may be numerically a higher value than the end coordinate.

A major difference with other proposed standards is that we chose to make strandedness and reference sequence a property of the position instead of the region.
This is important in a number of use cases.
One is when we someone needs to describe the position of a Gene on a rough assembly where the start and end are known to be on a different contig. 
This can be the case when RNA mapping is used in the genome assembly process.
Another is when rough semantics are used in queries e.g. answering what is the start and end of a Gene. 
In a process called transplicing exons of one gene can be found on multiple chromosomes e.g. \textit(TODO find an example)
In such a case the start of the gene is on a different reference sequence than the end.
We encourage that data providers describe the biology more precise than gene start-end in their databases, but we know that legacy data is not easily converted.
Another is Disulfide bonds in a protein complex the start position is on one chain while the end is on another chain in the complex.
FALDO can describe the position of the bridge with InBetweenPositions where the start and end of the bridge is on different protein chains.
This would not be possible if the reference sequence was a property of the region. 
The last difference is that it makes queries more predictable. 
Current formats describe single amino acid binding sites as regions.
FALDO allows us to describe such sites directly on positions therefore the Position class needs the reference sequence property.
For the Region class it is not required as a Region always ends and starts with Positions.

\subsection*{OWL2 reasoning based compression}
For large databases such as INDSC or UniProt the need to repeat the reference sequence for positions again and again incurs a significant cost in storage.
OWL2 reasoning introduced a new option called PropertyChain reasoning. 
With the following axioms the faldo:reference triples can be inferred for any position described by a INSDC record.
\begin{verbatim}
INDSC:reference
      a       owl:ObjectProperty ;
      rdfs:subPropertyOf faldo:reference ;
      owl:propertyChainAxiom
              (faldo:beginOf faldo:endOf :featureOf insdc:sequence) .
\end{verbatim}
A OWL capable query rewriter (e.g. such as part of Stardog) means that users do not see the difference between encoding the faldo:reference explicitly or having them inferred at query time.
Other stores that do not have such capabilities can easily add the necessary triples using a single SPARQL insert query.
\begin{verbatim}
CONSTRUCT {
    ?position faldo:reference ?sequence .
}
WHERE {
    ?record a insdc:Entry .
    ?record insdc:feature ?feature .
    ?feature faldo:position ?position .
    ?record insdc:sequence ?sequence .
}
\end{verbatim} 
For a SPARQL mapping to legacy files such as SPARQL-bed these extra triples do not matter as they do not exist in any way on disc .


FALDO has been designed to be open to extension for future
use \textit{TODO - expand on this with Jerven}.
