\section*{Discussion}
When designing FALDO, a broad range of use cases were considered from
human genome annotations to protein domains and glycan binding sites on
amino acid sequences, with the goal of developing a scheme general enough
to describe regions of DNA, RNA and protein sequences.

Advantages and drawbacks of existing file formats were considered, including
line based column formats like BED and GTF/GFF3 which focus on exact
ranges on a given sequence, and the more complex locations supported
by the INSDC feature tables used by DDBJ, NCBI-GenBank and EMBL-Bank.

The simplest non-stranded range location on a linear sequence requires
a start and end coordinate, but even here there are existing competing
conventions for describing open or closed end-points using zero and
one-based counting (for example BED versus GTF/GFF3/INSDC).

In FALDO we always count from the start of the forward 5'--3' strand, even for features on the reverse strand.
This encoding means there is no need to know the length of the sequence to compare positions on the different strands.
The end and start position of a region is inclusive.
Unlike formats like GTF/GFF3, FALDO shares with Chado\cite{Chado2007} the convention
that the start coordinate should be the biological start
(which may be a numerically higher value than the end coordinate).

%Similarly multiple schemes exist for describing strand specific locations,
%with some formats describing features on the reverse strand implicitly
%when the start coordinate is numerically higher than the end coordinate
%\textit{(TODO - example of format which does this)}.


For a semantic description describing the strand explicitly is preferable.
FALDO chooses to add the strand information to the position. 
This is required to accurately describe for example the sticky ends
of an enzyme digestion cut site, as in the HindIII example (Figure~\ref{fig:HindIII}).

A major difference with other standards is that we chose to make strandedness and reference sequence a property of the position, instead of the region.
This is important in a number of use cases.
For example, one may need to describe the position of a gene on a draft genome assembly where the start and end are known to be on different contigs.
This can be the case when RNA mapping is used in the genome assembly process.
Another is when rough semantics are used in queries e.g. answering what is the start and end of a gene. 
In a process called transplicing, exons of one gene can be found on multiple chromosomes, or on different strands of the same chromosome.
e.g.  gene \textit{mod}(\textit{mdg4}) of \textit{Drosophila melanogaster} (uniprot:Q86B87).
In such cases the start of the gene can be on a different reference sequence or strand than the end.
These biological realities cannot be described accurately if the reference sequence was a property of the region. 
As a side effect, it allows single nucleotide or amino acid sites to be described directly as a position without a need for an artificial region of length one.

%Another use case is describing disulfide bonds in a protein complex where the start position is on one chain while the end is on another chain in the complex.
%FALDO can describe the position of the disulfide bond with $\mathtt{faldo\colon{}InBetweenPositions}$ where the start and end of the bridge is on different protein chains.
%TODO find a better example than disulfide bonds for inbetween as the bond is a feature not a position.

Every $\mathtt{faldo\colon{}Position}$ refers to the sequence it is on.
This allows us to say that gene $XX$ starts at position 4 of assembly $Y1$,
while the same conceptual gene starts at position 5 of assembly $Y2$.
Even within the same assembly, FALDO offers the possibility to describe features in different contexts at the same time, allowing for instance to represent a SNP in terms of its position within a known coding region (i.e. gene coordinates) and within a chromosome region, which offers clear advantages for features annotation.
Chado also allows multiple locations per feature, but unlike FALDO,
the start and end of any location must be in the same region,
which prohibits for example a feature that spans more than one contig,
or describing the same feature on two different genome assemblies.

\subsection*{Efficiency of Region-of-Interest queries}

For FALDO we also considered query efficiency in comparison to existing search technology. 
Region of interest (ROI) queries are common operations performed on a
set of genome annotations to extract a set of features within a range.
For applications such as genome browsers, it is important that these are as efficient enough.
Although some RDF query engines may perform poorly when performing ROI queries over large feature sets,
others have special indexes (e.g. literal filter indexes) that improve query performance.
There is scope for further optimisation in the context of a SPARQL query by combining
efficient algorithms and indexes such as Nested Containment Lists (NCLs) \cite{NCL2007}
or spatial indexes.
%%Still unhappy with this section. Our modelling did not really look into it, and efficiency depends on store choices.s 

%\subsection*{Verbosity of FALDO}

%Its clear to anyone reading this paper that FALDO is considerably more verbose than existing formats.
%Once stored in an appropriately compressed file a RDF/FALDO representation does not need to take more space than a legacy format.
%In the case of UniProt, approximately 40\% of all data has been stated at least twice in the flat-file an XML representation.
%While the straight UniProt flatfile/xml translation into RDF consumes about 2 times the disk space, a unique triple
set of UniProt is currently only 5\% larger in size.
%%TODO Verify information once Matterhorn is restarted.
%This information does not need to be repeated for RDF consuming tools.
%SPARQL-BED shows that a legacy format such as BAM can be converted on demand to RDF and be used for SPARQL without needing more disk space to store sequencing data.

FALDO being a RDF based format can used to represent feature position information in a wide variety of serialisations e.g. JSON-LD, RDF/XML, Turtle, RDFa (embedded in HTML) this allows different developers use FALDO information in consideration of their usage scenario.
While at the same time allowing conversion to the common RDF triple model that can be used in RDF databases and accessed by SPARQL queries.


