%\documentclass{llncs}
%\NeedsTeXFormat{LaTeX2e}[1995/12/01]
\documentclass[10pt]{bmc_article}
\usepackage{cite} % Make references as [1-4], not [1,2,3,4]
\usepackage{url}  % Formatting web addresses  
\usepackage{ifthen}  % Conditional 
\usepackage{multicol}   %Columns
\usepackage[utf8]{inputenc} %unicode support
\usepackage{listings}
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails
\urlstyle{rm}


\def\includegraphic{}
\def\includegraphics{}



\setlength{\topmargin}{0.0cm}
\setlength{\textheight}{21.5cm}
\setlength{\oddsidemargin}{0cm} 
\setlength{\textwidth}{16.5cm}
\setlength{\columnsep}{0.6cm}

\newboolean{publ}
\newenvironment{bmcformat}{\begin{raggedright}\baselineskip20pt\sloppy\setboolean{publ}{false}}{\end{raggedright}\baselineskip20pt\sloppy}

\lstdefinelanguage[1.1]{sparql}{morekeywords={PREFIX,SELECT,CONSTRUCT,WHERE,ASK,DESCRIBE,A,NOT,EXISTS,FILTER,regex},sensitive=true,morestring=[s]{<>},morestring=[s]{""},escapeinside={~}{~},extendedchars=true,morecomment=[l]{\#}}
\lstset{
  basicstyle=\small, % print whole listing small
  identifierstyle=, % nothing happens
  commentstyle=\color{blue}, % white comments
  stringstyle=\ttfamily, % typewriter type for strings
  showstringspaces=false
  numbers=left, 
  numberstyle=\tiny,
  numberbychapter=true,
  xleftmargin=0.1\linewidth,
  xrightmargin=0.1\linewidth,
  boxpos=c,
  frame=, 
  numbersep=2p,
  columns=fullflexible,
  escapechar=`
}

%\newfloat{example}{h}{ttl}[section]
%\floatname{example}{Example}		
\begin{document}
\begin{bmcformat}
\title{Semantic standard for describing the location of nucleotide and protein feature annotation.}
\author{Jerven Bolleman\correspondingauthor$^1$ \email{Jerven Bolleman\correspondingauthor - jerven.bolleman@isb-sib.ch} 
         \and Robert Beuels$^2$%
         \and Robert Hoehndorf$^3$%
         \and Peter Cock$^3$ %
		 \and Toshiaki Katayama$^4$ %
		 \and Raoul JP Bonnal$5$ %
		 \and Michel Dumontier$^5$ %
		 \and Takatomo Fujisawa$^6$ %
		 \and Francesco Strozzi$^7$ %
		 \and Joachim Baran$^8$ %
         }
\address{
    \iid(1)SIB Swiss Institute of Bioinformatics, Centre Medical Universitaire, 1 rue Michel
Servet, 1211 Geneva 4, Switzerland,
 \iid(2) fill in,
 \iid(3) James Hutton Institute, Invergowrie, Dundee DD2 5DA, UK,
 \iid(4) fill in,
 \iid(5) Integrative Biology Program, Istituto Nazionale Genetica Molecolare, Milan, Italy,
 \iid(6) fill in,
 \iid(7) fill in, and
 \iid(8) fill in.
        }
\maketitle

\input{abstract}
\section*{Keywords}

\input{background}
\input{implementation}
\input{results}
\input{discussion}
\input{conclusions}
\input{avareq}

\section*{List of abbreviations}
\begin{description}
\item[SPARQL] SPARQL Protocol and RDF Query Language
\item[UniProtKB] Universal Protein Knowledgebase 
\item[RDF] Resource Description Framework
\item[DNA] Deoxyribonucleic acid
\end{description}
\bigskip

\section*{Authors contributions}

Jerven Bolleman wrote the basic ontology file and mapping to UniProt RDF.
Robert Hoehndorf did the GFF3 to OWL conversion.
Peter Cock adapted the Biopython data object model to this format (TODO).
Raoul JP Bonnal adapted BioRuby to match the ontology and wrote the cufflink to locations.rdf converter with Francesco Strozzi.

Totokama Fujisawa implemented the DDBJ RDF in terms of this format.
Robert Beuels adapted JBrowse to query SPARQL endpoints that use this format to generate custom tracks. 

\section*{Acknowledgments}



\newpage
{\ifthenelse{\boolean{publ}}{\footnotesize}{\small}
 \bibliographystyle{bmc_article}  % Style BST file
  \bibliography{locations} }     % Bibliography file (usually '*.bib' ) 
  
\end{bmcformat}
\end{document}
