%\documentclass{llncs}
%\NeedsTeXFormat{LaTeX2e}[1995/12/01]
\documentclass[10pt]{bmc_article}
\usepackage{cite} % Make references as [1-4], not [1,2,3,4]
\usepackage{url}  % Formatting web addresses  
\usepackage{ifthen}  % Conditional 
\usepackage{multicol}   %Columns
\usepackage[utf8]{inputenc} %unicode support
\usepackage{listings}
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails
\urlstyle{rm}


\def\includegraphic{}
\def\includegraphics{}



\setlength{\topmargin}{0.0cm}
\setlength{\textheight}{21.5cm}
\setlength{\oddsidemargin}{0cm} 
\setlength{\textwidth}{16.5cm}
\setlength{\columnsep}{0.6cm}

\newboolean{publ}
\newenvironment{bmcformat}{\begin{raggedright}\baselineskip20pt\sloppy\setboolean{publ}{false}}{\end{raggedright}\baselineskip20pt\sloppy}

\lstdefinelanguage[1.1]{sparql}{morekeywords={PREFIX,SELECT,CONSTRUCT,WHERE,ASK,DESCRIBE,A,NOT,EXISTS,FILTER,regex},sensitive=true,morestring=[s]{<>},morestring=[s]{""},escapeinside={~}{~},extendedchars=true,morecomment=[l]{\#}}
\lstset{
  basicstyle=\small, % print whole listing small
  identifierstyle=, % nothing happens
  commentstyle=\color{blue}, % white comments
  stringstyle=\ttfamily, % typewriter type for strings
  showstringspaces=false
  numbers=left, 
  numberstyle=\tiny,
  numberbychapter=true,
  xleftmargin=0.1\linewidth,
  xrightmargin=0.1\linewidth,
  boxpos=c,
  frame=, 
  numbersep=2p,
  columns=fullflexible,
  escapechar=`
}

%\newfloat{example}{h}{ttl}[section]
%\floatname{example}{Example}		
\begin{document}
\begin{bmcformat}
\title{Semantic standard for describing the location of nucleotide and protein feature annotation.}
\author{Jerven Bolleman\correspondingauthor$^1$ \email{Jerven Bolleman\correspondingauthor - jerven.bolleman@isb-sib.ch} 
         \and Robert Beuels$^2$%
         \and Robert Hoehndorf$^3$%
         \and Peter Cock$^3$ %
		 \and Toshiaki Katayama$^4$ %
		 \and Raoul Bonnal$5$ %
		 \and Michel Dumontier$^5$ %
		 \and Takatomo Fujisawa$^6$ %
		 \and Francesco Strozzi$^7$ %
		 \and Joachim Baran$^8$ %
         }
\address{
    \iid(1)SIB Swiss Institute of Bioinformatics, Centre Medical Universitaire, 1 rue Michel
Servet, 1211 Geneva 4, Switzerland,
 \iid(2) fill in,
 \iid(3) fill in,
 \iid(4) fill in,
 \iid(5) INGM,
 \iid(6) fill in,
 \iid(7) fill in, and
 \iid(8) fill in.
        }
\maketitle

\begin{abstract}
\begin{description}
\item[Background] Nucleotide and protein sequence feature annotation is essential to understand our biology on the genome, transcriptome, and proteome level. Unfortunately, there is no standard that describes the describes this potentially complex location information as
subject-predicate-object triples suitable for semantic web queries using RDF/SPARQL.
\item[Description] We developed a schema ontology to describe the positions of annotated features that is usable for describing glycan binding sites, protein annotation and nucleotide features in sequence records.
Using the same data format to represent sequence positions independent of files allows us to integrate sequence data from multiple sources and types.
The integration capabilities are shown by JBrowse using multiple SPARQL endpoints to display genomic feature annotation as well as protein annotation from UniProt mapped to genomic locations.
\item[Conclusions] This standard ontology allows users to merge sequence annotation from multiple sources using federalized SPARQL queries against public endpoints and local private datasources.
\end{description}
\end{abstract}

\section{Keywords}
SPARQL, RDF, semantic-web, standardization, sequence ontology

\section{Background}
Understanding how nucleotide and protein sequences in databases correspond to the real world is key to understanding biological processes.
Determined nucleotide sequences are stored in a large number of databases, such as GenBank/DDBJ/EMBL and vast numbers of new sequence records are generated every day since the advent of high throughput sequencing. Currently there is a vast number of ways of describing such sequences and their biological relation e.g. GenBank format, GFF3 and variants. All of these formats require the annotation to be stored in the same file as the sequence, making it difficult to merge information between files without copying all data.

While determining sequences has become vastly cheaper over time basic curation of annotation has not. This means that we still move a most annotation from one sequence to another, e.g. gene annotation moves from one genome assembly to the next.

The proposed standard allows us to accurately describe the position of a feature on multiple sequences. For example a gene can start at position X on genome assembly A while the same gene is positioned on X' on genome assembly A+1. The proposed standard also deals with incomplete assemblies as well as protein sequence annotation such as found in UniProtKB.

\section{List of abbreviations}
\begin{description}
\item[SPARQL] SPARQL Protocol and RDF Query Language
\item[UniProtKB] Universal Protein Knowledgebase 
\item[RDF] Resource Description Framework
\item[DNA] Deoxyribonucleic acid
\end{description}
\bigskip

\section{Authors contributions}

Jerven Bolleman wrote the basic ontology file and mapping to UniProt rdf. Robert Hoehndorf did the gff3 to OWL conversion. Peter Cock adapted the BioPython data object model to this format. Raoul Bonnal adapted BioRuby to match the ontology. Fransesco Strozzi wrote the cufflink to locations.rdf converter. Totokama Fujisawa. implemented the DDBJ rdf in terms of this format. Robert Beuels adapted JBrowse to query sparql endpoints that use this format to generate custom tracks. 

\section{Acknowledgments}



\newpage
{\ifthenelse{\boolean{publ}}{\footnotesize}{\small}
 \bibliographystyle{bmc_article}  % Style BST file
  \bibliography{locations} }     % Bibliography file (usually '*.bib' ) 
  
\end{bmcformat}
\end{document}