%\documentclass{llncs}
%\NeedsTeXFormat{LaTeX2e}[1995/12/01]
\documentclass[10pt]{bmc_article}
\usepackage{cite} % Make references as [1-4], not [1,2,3,4]
\usepackage{url}  % Formatting web addresses  
\usepackage{ifthen}  % Conditional 
\usepackage{multicol}   %Columns
\usepackage[utf8]{inputenc} %unicode support
\usepackage{listings}

%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails
\urlstyle{rm}

%Shaded framed
\usepackage{framed,color}
\definecolor{shadecolor}{rgb}{0.96,0.96,0.96}

%\def\includegraphic{}
%\def\includegraphics{}
\usepackage{graphicx}

\setlength{\topmargin}{0.0cm}
\setlength{\textheight}{21.5cm}
\setlength{\oddsidemargin}{0cm} 
\setlength{\textwidth}{16.5cm}
\setlength{\columnsep}{0.6cm}

\newboolean{publ}
\newenvironment{bmcformat}{\begin{raggedright}\baselineskip20pt\sloppy\setboolean{publ}{false}}{\end{raggedright}\baselineskip20pt\sloppy}

\lstdefinelanguage[1.1]{sparql}{morekeywords={PREFIX,SELECT,CONSTRUCT,WHERE,ASK,DESCRIBE,A,NOT,EXISTS,FILTER,regex},sensitive=true,morestring=[s]{<>},morestring=[s]{""},escapeinside={~}{~},extendedchars=true,morecomment=[l]{\#}}
\lstset{
  basicstyle=\small, % print whole listing small
  identifierstyle=, % nothing happens
  commentstyle=\color{blue}, % white comments
  stringstyle=\ttfamily, % typewriter type for strings
  showstringspaces=false
  numbers=left, 
  numberstyle=\tiny,
  numberbychapter=true,
  xleftmargin=0.1\linewidth,
  xrightmargin=0.1\linewidth,
  boxpos=c,
  frame=, 
  numbersep=2p,
  columns=fullflexible,
  escapechar=`
}

%\newfloat{example}{h}{ttl}[section]
%\floatname{example}{Example}		
\begin{document}
\begin{bmcformat}
\title{Semantic standard for describing the location of nucleotide and protein feature annotation.}
\author{Jerven Bolleman\correspondingauthor$^{1}$\email{Jerven Bolleman\correspondingauthor - jerven.bolleman@isb-sib.ch}%
    \and Christopher J Mungall$^{2}$%
    \and Francesco Strozzi$^{3}$%
    \and Joachim Baran$^{4}$%
    \and Michel Dumontier$^{5}$%
    \and Raoul J. P. Bonnal$^{6}$%    
    \and Robert Buels$^{7}$%
    \and Robert Hoehndorf$^{8}$%
    \and Takatomo Fujisawa$^{9}$%
    \and Toshiaki Katayama$^{10}$%
    \and Peter J. A. Cock$^{11}$%
%
}
\address{
 \iid(1) SIB Swiss Institute of Bioinformatics, Centre Medical Universitaire, 1 rue Michel
Servet, 1211 Geneva 4, Switzerland,
 \iid(2) Genomics Division, Lawrence Berkeley National Laboratory, Berkeley, CA, 94720, US,
 \iid(3) CeRSA, Parco Tecnologico Padano, Lodi 26900, Italy,
 \iid(4) Ontario Institute for Cancer Research, 101 College Street, Suite 800, Toronto, Ontario, M5G 0A3, Canada,
 \iid(5) Stanford Center for Biomedical Informatics Research, 1265 Welch Road, Room X223, Stanford, CA, 94305-5479, US,
 \iid(6) Integrative Biology Program, Istituto Nazionale Genetica Molecolare, Milan, Italy,
 \iid(7) University of California, Berkeley, Berkeley, CA, USA,
 \iid(8) Department of Computer Science, Aberystwyth, SY23 3DB, UK,
 \iid(9) Center for Information Biology, National Institute of Genetics, Research Organization of Information and Systems, 1111 Yata, Mishima, Shizuoka 411-08540, Japan,
 \iid(10) Database Center for Life Science, Research Organization of Information and Systems, 2-11-16, Yayoi, Bunkyo-ku, Tokyo, 113-0032, Japan, and
 \iid(11) The James Hutton Institute, Dundee, DD2 5DA, UK.
}
\maketitle

\input{abstract}
\section*{Keywords}
SPARQL, RDF, semantic-web, standardisation, sequence ontology, annotation, data integration

\input{background}
\input{implementation}
\input{results}
\input{discussion}
\input{conclusions}
\input{avareq}


\section*{List of abbreviations}

\begin{description}
\item[SPARQL] SPARQL Protocol and RDF Query Language
\item[UniProtKB] Universal Protein Knowledgebase 
\item[RDF] Resource Description Framework
\item[DNA] Deoxyribonucleic acid
\item[INSDC] International Nucleotide Sequence Database Collaboration
\item[DDBJ] DNA Data Bank of Japan
\item[EMBL] European Molecular Biology Laboratory
\item[ENA] European Nucleotide Archive
\item[BED] Browser Extensible Data (file format)
\item[GTF] Gene Transfer Format
\item[GFF] Generic Feature Format
\item[GFF3] Generic Feature Format version 3
\item[OWL] Web Ontology Language (note acronym is OWL, not WOL)
\end{description}
\bigskip

\section*{Authors contributions}

Jerven Bolleman wrote the basic ontology file and mapping to UniProt RDF and contributed to the text of this article.
Robert Hoehndorf did the GFF3 to OWL conversion.
Peter Cock wrote a large section of this paper.
Raoul JP Bonnal adapted BioRuby to match the ontology and wrote the cufflink to locations.rdf converter with Francesco Strozzi.
Takatomo Fujisawa and Toshiaki Katayama implemented the DDBJ RDF in terms of this format and used FALDO to describe features on bacterial genomes in TogoGenome.
Robert Buels adapted JBrowse to query SPARQL endpoints that use this format to generate custom tracks. 

\section*{Acknowledgments}
The developers recognize the invaluable contributions from the community in helping to create this standard.
This work has been supported by National Bioscience Database Center (NBDC) of Japan Science and Technology Agency (JST) and Database Center for Life Science (DBCLS) of Research Organization of Information and Systems (ROIS) in Japan.
Jerven Bolleman was supported in his role at Swiss-Prot, whose activities at the SIB Swiss Institute of Bioinformatics are supported by the Swiss Federal Government through the The State Secretariat for Education, Research and Innovation SERI.
Christopher Mungall was supported by the NIH under R24OD011883 and by the Director, Office of Science, Office of Basic Energy Sciences, of the U.S. Department of Energy under Contract No. DE-AC02-05CH11231.
Peter Cock was supported by the Scottish Government Rural and Environmental Research and Analysis Directorate.
Takatomo Fujisawa was supported by the DNA Databank of Japan (DDBJ), Research Organization of Information and Systems (ROIS) in Japan.

\newpage
{\ifthenelse{\boolean{publ}}{\footnotesize}{\small}
 \bibliographystyle{bmc_article}  % Style BST file
  \bibliography{locations} }     % Bibliography file (usually '*.bib' ) 
  
\end{bmcformat}
\end{document}
