\section*{Results}
One of the practical goal driving the development of FALDO was to be able
to represent all the annotated sequences in UniProt and the INSDC as RDF
triples, as a step towards providing this data via SPARQL end points whereby
it can be queried.
UniProt is already available as RDF, and a revised version of the full
dataset using FALDO has already tested and is expected to be included
in the UniProt XXX release \textit{TODO - fill in version and data}.

The protein examples considered, such as the UniProt feature annotations,
require relatively simple locations within protein sequences to be described.
There are corner cases for situations like unknown boundaries.
\textit{TODO - examples - how about cysteine bonds in PDB data?}.

Feature locations on nucleotide sequences can be relatively simple,
%GenBank
for example the \textit{cheY} gene in
\textit{Escherichia coli} str. K-12 substr. MG1655 (accession NC\_000913.2)
is described in the INSDC feature table as \texttt{complement(1965072..1965461)},
which is 390 base pairs using inclusive one-based counting.
This is interpreted as \texttt{complement($end$..$start$)}, where
the feature begins on the base complementary to $start = 1965461$
and finishes at $end = 1965072$. FALDO respects this biological
interpretation of a feature location on the reverse strand:

%Notice begin, end = 1965461, 1965072 (biological interpretation)
\begin{verbatim}
<_:geneCheY> a so:Gene ;
           rdfs:label "cheY" ;
           faldo:location <_:example> ;

uniprot:P0AE67 up:encodedBy <_:geneCheY> .

<_:example> a faldo:Region ;
           faldo:begin <_:example_b> ;
           faldo:end <_:example_e> .

<_:example_b> a faldo:Position ; 
           a faldo:ExactlyKnownPosition ;
           a faldo:ReverseStrandPosition ;
           faldo:position "1965461"^^xsd:int ;
           faldo:reference refseq:NC_000913.2 .

<_:example_e> a faldo:Position ; 
           a faldo:ExactlyKnownPosition ;
           a faldo:ReverseStrandPosition ;
           faldo:position "1965072"^^xsd:int ;
           faldo:reference refseq:NC_000913.2 .
\end{verbatim}

In contrast, other formats like GFF require $start \leq end$
regardless of the strand, which is equivalent to interpreting
the INSDC location string as \texttt{complement($start$..$end$)}.
This convention has a number of practical advantages when
dealing with numerical operations on features sets, such as
checking for overlaps or indexing data. For example, the
feature length is given by $length = end - start + 1$ under
this numerically convenient scheme where the interpretation
of $start$ versus $end$ is strand independent.

Nucleotide data from the INSDC can require considerably more complicated
sequence regions, both in terms of ambiguous end points, and the need for
various compound locations.

\textit{TODO - examples}.

All of the UniProt protein sequences are now available in RDF format,
with a SPARQL end point currently in beta. Converting more recent
curated INSDC data has been straightforward, however there are a
number of deprecated location strings which have been problematic.
\textit{TODO - examples? manual curation to fix last few oddities?}
