\section*{Results}
One of the practical goal driving the development of FALDO was to be able
to represent all the annotated sequences in UniProt and the INSDC as RDF
triples, as a step towards providing this data via SPARQL end points whereby
it can be queried.

The protein examples considered, such as the UniProt feature annotations,
require relatively simple locations within protein sequences to be described (see the active site annotation in figure \ref{fig:UniProtDDBJ}).

\begin{figure}
\begin{shaded}
\begin{verbatim}
AC   Q6Q250;
...
DE   Flags: Precursor; Fragment;
...
FT   SIGNAL       <1     15
FT   ACT_SITE    153    153       Nucleophile (By similarity).
\end{verbatim}
\begin{verbatim}
uniprot:Q6Q250 rdf:type :Protein ;
   up:annotation SHA:E50..1B ;
   up:sequence isoform:P05064-1 ;
   rdfs:seeAlso ddbj-cds:AAS67043.1 .
SHA:E50..1B a up:Signal_Peptide_Annotation ;
  faldo:location region:22859195476160557tf1tt15> .
SHA:E50..1C a up:Active_Site_Annotation ;
  faldo:location position:22859195476160557tf153tt153> .
region:22859195476160557tf1tt15 a faldo:Region ;
   faldo:begin position:before1 ;
   faldo:end  position:15
position:before1 a faldo:InRangePosition ;
   faldo:reference isoform:P05064-1 ;
   faldo:end position:1 . 
position:1 a faldo:ExactPosition ;
   faldo:reference isoform:P05064-1 ;
   faldo:position 1 .
position:15 a faldo:ExactPosition ;
   faldo:reference isoform:P05064-1 ;
   faldo:position 15 .   
position:153 a faldo:ExactPosition ;
   faldo:reference isoform:P05064-1 ;
   faldo:position 153 .
isoform:P05064-1 up:fragment "single" ;
   up:precursor true .
\end{verbatim}
\begin{verbatim}
LOCUS       AY566647                 948 bp    mRNA    linear   INV 22-MAR-2004
...
FEATURES             Location/Qualifiers
...
     CDS             <1..>948
                     /note="allergen Pol d 1.03"
                     /codon_start=1
                     /product="venom phospholipase A1 3 precursor"
                     /protein_id="AAS67043.1"
                     /db_xref="GI:45510891"
                     /translation="ADDLTTLRNGTLDRGITPDCTFNEKDIELHVYSRDKRNGIILKK
                     EILKNYDLFQKSQISHQIAILIHGFLSTGNNENFDAMAKALIEIDNFLVISVDWKKGA
                     CNAFASTNDVLGYSQAVGNTRHVGKYVADFTKLLVEQYKVPMSNIRLIGHSLGAHTSG
                     FAGKEVQRLKLGKYKEIIGLDPAGPSFLTSKCPDRLCETDAEYVQAIHTSAILGVYYN
                     VGSVDFYVNYGKSQPGCSEPSCSHTKAVKYLTECIKRECCLIGTPWKSYFSTPKPISQ
                     CKRDTCVCVGLNAQSYPAKGSFYVPVDKDAPYCHNEGIKL"
\end{verbatim}
\begin{verbatim}
ddbj-cds:AAS67043.1 rdf:type :Nucleotide_Resource ;
  faldo:location ddbj-reg:AY566647b1a948 .
ddbj-reg:AY566647b1a948 a faldo:Region ;
	faldo:begin ddbj-reg:AY566647before1 ;
	faldo:end ddbj-reg:AY566647after948 .
ddbj-reg:AY566647before1 a faldo:InRangePosition ;
	faldo:before ddbj-reg:AY566647p1 .
ddbj-reg:AY566647after948 a faldo:InRangePosition ;
	faldo:after ddbj-reg:AY566647p948 ;
ddbj-reg:AY566647p1	a faldo:ExactPosition ;
	faldo:position 1 ;
	faldo:reference ddbj-seq:AY566647 .
ddbj-reg:AY566647p948 a faldo:ExactPosition ;
	faldo:position 948 ;
	faldo:reference ddbj-seq:AY566647 .
\end{verbatim}
\end{shaded}
\caption{Example from UniProt entry P05064 showing the position of an active site in both faldo and the UniProt flat-file format. As well as the underlying DDBJ record showing the related CDS sequence and how its coding region start and ends outside of the known deposited mRNA sequence.}
\label{fig:UniProtDDBJ}
\end{figure}



%
%Notice begin, end = 1965461, 1965072 (biological interpretation)
\begin{figure}
\begin{shaded}
\small
\begin{verbatim} 
@prefix faldo: <http://biohackathon.org/resource/faldo#> .

# Here are example triples describing a gene, note the location line:
<_:geneCheY> a so:Gene ;
           rdfs:label "cheY" ;
           faldo:location <_:example> ;

# The following triples define the location used by the feature above,
# made up of a region and its two positions:
<_:example> a faldo:Region ;
           faldo:begin <_:example_b> ;
           faldo:end <_:example_e> .

<_:example_b> a faldo:Position ; 
           a faldo:ExactlyKnownPosition ;
           a faldo:ReverseStrandPosition ;
           faldo:position "1965461"^^xsd:int ;
           faldo:reference refseq:NC_000913.2 .

<_:example_e> a faldo:Position ; 
           a faldo:ExactlyKnownPosition ;
           a faldo:ReverseStrandPosition ;
           faldo:position "1965072"^^xsd:int ;
           faldo:reference refseq:NC_000913.2 .
\end{verbatim}
\end{shaded}
\caption{Incomplete example of using Faldo in turtle format to describe
gene chrY being at complement(NC\_$000913.2:1965072..1965461$)}
\label{fig:insdcComplement}
\end{figure}


\subsection*{Complement strand and ISDC compound locations}

Describing biological features in relation to a genomic DNA sequence does not have to be complicated.

For example the \textit{cheY} gene (shown in Fig:\ref{fig:insdcComplement}) 
\textit{Escherichia coli} str. K-12 substr. MG1655 (accession NC\_000913.2)
is described in the INSDC feature table as \texttt{complement(1965072..1965461)},
which is 390 base pairs using inclusive one-based counting.
This feature begins on the base complementary to $start = 1965461$
and finishes at $end = 1965072$, and so the INSDC location string
can be interpreted as \texttt{complement($end$..$start$)},
FALDO respects this biological interpretation of a feature location
on the reverse strand:

In contrast, other formats like GFF require $start \leq end$
regardless of the strand, which is equivalent to interpreting
the INSDC location string as \texttt{complement($start$..$end$)}.
This convention has some practical advantages when
dealing with numerical operations on features sets, such as
checking for overlaps or indexing data. For example, the
feature length is given by $length = end - start + 1$ under
this numerically convenient scheme where the interpretation
of $start$ versus $end$ is strand independent.

However, there are a number of implicit conventions in INSDC data that need to be translated into the simpler FALDO world.
Some of the complicated regions for INSDC are features on a circular genome. 
The most common is  a feature that straddles the origin of replication or the '0' mark.
For example a the "Protein II" gene encoded on the reverse strand of the f1 bacteriophage (ddbj:J02448) that straddles the origin of replication.
"Protein II" transcription starts from reverse strand position 6006 and the transcription ends at reverse strand position  831 (see figure \ref{fig:insdcOverOrigin}).


\begin{figure}
\begin{shaded}
\small
\begin{verbatim}
# 
@prefix faldo: <http://biohackathon.org/resource/faldo#> .
@prefix ddbj-.... : examples of possible DDBJ/INSDC related prefixes
# Here are example triples describing a gene, note the location line:
ddbj-record:J02448 dc:identifier "J02448" ;
           rdfs:Comment "Bacteriophage f1, complete genome" ;
           taxonomy:10863 ;
           ddbj:CDS ddbj ddbj-protein:AAA32209.1 .
ddbj-protein:AAA32209.1 a so:0000316 ; #CDS
           rdfs:label "Protein II" ;
           faldo:location ddbj-reg:J02448f6006t831 ;

ddbj-reg:J02448f6006t831 a faldo:Region ;
           faldo:begin ddbj-pos:J02448p6006+ ;
           faldo:end ddbj-pos:J02448p831+ .

ddbj-pos:J02448p6006+ a faldo:Position ,
             faldo:ExactlyKnownPosition ,
             faldo:ForwardStrandPosition ;
           faldo:position 6006 ;
           faldo:reference ddbj-seq:J02448 .

ddbj-pos:J02448p831+ a faldo:Position , 
             faldo:ExactlyKnownPosition ,
             faldo:ForwardStrandPosition ;
           faldo:position 831 ;
           faldo:reference ddbj-seq:J02448 .
           
\end{verbatim}
\end{shaded}
\caption{Incomplete example of using Faldo in turtle format to describe
CDS protein II being at J02448 $join(6006..6407,1..831)$}
\label{fig:insdcReverseOverOrigin}
\end{figure}


\subsection*{Fuzzy locations}
Nucleotide data from the INSDC or experimental data described in UniProt can also require ambiguous end points. 
For example position the Signal Peptide annotation shown in figure \ref{fig:UniProtDDBJ}, 
here the protein sequence is known to belong to a family of proteins, unfortunately only a part of the amino acid sequence is known. 
The UniProt curator deduced that the Signal Peptide region starts only partly overlaps the known sequence fragment.
The same is true in the related INSDC record, were the CDS starts and ends before the known mRNA sequence.

\subsection*{Restriction enzymes}

The recognition sites of most restriction enzymes are relatively
straightforward to describe, as is the cleavage site of a blunt
end cutting enzyme \textit{TODO - example of this? Between location}.

However, the cutting point of sticky-end cutting enzymes which
leave an overhang are more challenging to describe. Here the
flexibility in FALDO to allow the start and end positions of a region
to have different strands can be used (Figure~\ref{fig:HindIII}).

\begin{figure}
\begin{shaded}
\small
\begin{verbatim}
    \/
5'..AAGCTT..3' 
    123456
3'..TTCGAA..5'
        /\
        
:EnzymeRestrictionSite a example:EnzymeRestrictionSite ;
       faldo:location :EnzymeRestrictionRegion .
:EnzymeRestrictionRegion a faldo:Region ;
       faldo:begin :53_1 ;
       faldo:end   :35_6 .
:53cleaveageSite a example:CleavageSite ;
       faldo:location :53cleaveageSitePosition .
:35cleaveageSite a example:CleavageSite ;
       faldo:location :35cleaveageSitePosition .
:53cleaveageSitePosition a faldo:InBetweenPosition, 
                           SO:0001689  ; # 5' restriction enzyme junction 
       faldo:after :53_1 ;        
       faldo:before :53_2 .
:35cleaveageSitePosition a faldo:InBetweenPosition, 
                           SO:0001690 ; # 3' restriction enzyme junction 
       faldo:before :35_5 ;
       faldo:after :35_6 . #Note the before after reversion due to biological start      
:53_1 a faldo:ExactPosition , faldo:ForwardStrandedPosition ;
      faldo:position 1 .
:53_2 a faldo:ExactPosition , faldo:ForwardStrandedPosition ;
      faldo:position 2 .
:35_5 a faldo:ExactPosition , faldo:ReverseStrandedPosition ;
      faldo:position 5 .
:35_6 a faldo:ExactPosition , faldo:ReverseStrandedPosition ;
      faldo:position 6 .
\end{verbatim}
\end{shaded}
\caption{HindIII restriction enzyme cleavage site with sticky ends}
\label{fig:HindIII}
\end{figure}


%\subsection*{More examples}

%\textit{TODO - other ideas:}
%\begin{itemize}
%\item Example of Cys bonding between one or two proteins?
%\item RNA/DNA digest sites, and the cut sites -- especially for over hangs (start and ends on opposite strands)
%\item tRNA hairpin (two complementary regions)?
%\item $\beta$-sheet in protein?
%\item RNA-Seq mapping (useful for SAM/BAM)?
%\end{itemize}


