\section*{Results}
One of the practical goals driving the development of FALDO was to be able
to represent all the annotated sequences in INSDC and UniProt as RDF
triples, as a step towards providing this data via SPARQL endpoints where
it can be queried.

The protein examples considered here, such as the UniProt feature annotations,
describe relatively simple locations within protein sequences
(see the active site annotation in Figures~\ref{fig:UniProt} and \ref{fig:DDBJ}).

\begin{figure}
\begin{shaded}
\small
\begin{verbatim}
AC   Q6Q250;
...
DE   Flags: Precursor; Fragment;
...
FT   SIGNAL       <1     15
FT   ACT_SITE    153    153       Nucleophile (By similarity).
\end{verbatim}
\begin{verbatim}
uniprot:Q6Q250 rdf:type up:Protein ;
   up:annotation SHA:E50-1B, SHA:E50-1C ;
   up:sequence isoform:Q6Q250-1 ;
   rdfs:seeAlso ddbj-cds:AAS67043.1 .
SHA:E50-1B a up:Signal_Peptide_Annotation ;
  faldo:location region:f1t15 .
SHA:E50-1C a up:Active_Site_Annotation ;
  faldo:location position:153 .
region:f1t15 a faldo:Region ;
   faldo:begin position:before1 ;
   faldo:end  position:15 .
position:before1 a faldo:InRangePosition ;
   faldo:reference isoform:Q6Q250-1 ;
   faldo:end position:1 . 
position:1 a faldo:ExactPosition ;
   faldo:reference isoform:Q6Q250-1 ;
   faldo:position 1 .
position:15 a faldo:ExactPosition ; #End of the signal peptide region
   faldo:reference isoform:Q6Q250-1 ;
   faldo:position 15 .   
position:153 a faldo:ExactPosition ; #Where the active site should be
   faldo:reference isoform:Q6Q250-1 ;
   faldo:position 153 .
isoform:P05064-1 up:fragment "single" ;
   up:precursor true .
\end{verbatim}
\end{shaded}
\caption{Excerpt from UniProt entry Q6Q250 showing the position of an active site and a signal peptide in both the UniProt flat-file format and FALDO.}
\label{fig:UniProt}
\end{figure}
\begin{figure}
\begin{shaded}
\small
\begin{verbatim}
LOCUS       AY566647                 948 bp    mRNA    linear   INV 22-MAR-2004
...
FEATURES             Location/Qualifiers
...
     CDS             <1..>948
                     /note="allergen Pol d 1.03"
                     /codon_start=1
                     /product="venom phospholipase A1 3 precursor"
                     /protein_id="AAS67043.1"
                     /db_xref="GI:45510891"
                     /translation="ADDLTTLRNGTLDRGITPDCTFNEKDIELHVYSRDKRNGIILKK
                     EILKNYDLFQKSQISHQIAILIHGFLSTGNNENFDAMAKALIEIDNFLVISVDWKKGA
                     CNAFASTNDVLGYSQAVGNTRHVGKYVADFTKLLVEQYKVPMSNIRLIGHSLGAHTSG
                     FAGKEVQRLKLGKYKEIIGLDPAGPSFLTSKCPDRLCETDAEYVQAIHTSAILGVYYN
                     VGSVDFYVNYGKSQPGCSEPSCSHTKAVKYLTECIKRECCLIGTPWKSYFSTPKPISQ
                     CKRDTCVCVGLNAQSYPAKGSFYVPVDKDAPYCHNEGIKL"
\end{verbatim}
\begin{verbatim}
ddbj-cds:AAS67043.1 rdf:type :Nucleotide_Resource ;
  faldo:location [ a faldo:Region ;
                   faldo:begin [ a faldo:InRangePosition ;
                                 faldo:before [a faldo:ExactPosition ;
                                              faldo:position 1 ;
                                              faldo:reference ddbj-seq:AY566647 ]]];
                   faldo:end   [ a faldo:InRangePosition ;
                                 faldo:after [a faldo:ExactPosition ;
                                              faldo:position 948 ;
                                              faldo:reference ddbj-seq:AY566647 ]]].
\end{verbatim}
\end{shaded}
\caption{DDBJ record associated with UniProt Q6Q250 showing the related CDS sequence, with coding region outside of the known deposited mRNA sequence.}
\label{fig:DDBJ}
\end{figure}



%
%Notice begin, end = 1965461, 1965072 (biological interpretation)
\begin{figure}
\begin{shaded}
\small
\begin{verbatim} 
@prefix faldo:       <http://biohackathon.org/resource/faldo#> .
@prefix ddbj-record: <http://ddbj.nig.ac.jp/entry/nucleotide/> . 

# Here are example triples describing a gene, note the location line:
<_:geneCheY> a so:Gene ;
           rdfs:label "cheY" ;
           faldo:location <_:example> .

# The following triples define the location used by the feature above,
# made up of a region and its two positions:
<_:example> a faldo:Region ;
           faldo:begin [ a faldo:Position ,
                           faldo:ExactlyKnownPosition ,
                           faldo:ReverseStrandPosition ;
                         faldo:position 1965461 ;
                         faldo:reference ddbj-record:U00096.3 ];
           faldo:end  [ a faldo:Position ,
                          faldo:ExactlyKnownPosition ;
                          faldo:ReverseStrandPosition ;
                        faldo:position 1965072
                        faldo:reference ddbj-record:U00096.3 ].
\end{verbatim}
\end{shaded}
\caption{Using FALDO in Turtle\cite{TurtleFormatSpec} syntax to describe the location of a
gene feature \textit{cheY} at \texttt{complement(NC\_000913.2:1965072..1965461)} in a INSDC record U00096.3.}
\label{fig:insdcComplement}
\end{figure}


\subsection*{Complement strand and INSDC compound locations}

Describing biological features in relation to a genomic DNA sequence does not have to be complicated.

For example the \textit{cheY} gene (shown in Figure~\ref{fig:insdcComplement})
\textit{Escherichia coli} str. K-12 substr. MG1655 (accession NC\_000913.2)
is described in the INSDC feature table as \texttt{complement(1965072..1965461)},
which is 390 base pairs using inclusive one-based counting.
This feature begins on the base complementary to $start = 1965461$
and finishes at $end = 1965072$, so the INSDC location string
can be interpreted as \texttt{complement($end$..$start$)}.
FALDO respects this biological interpretation of a feature location
on the reverse strand.

In contrast, other formats such as the GFF family of formats, require $start \leq end$
regardless of the strand, which is equivalent to interpreting
the INSDC location string as \texttt{complement($start$..$end$)}.
This convention has some practical advantages when
dealing with numerical operations on features sets, such as
checking for overlaps or indexing data. For example, the
feature length is given by $length = end - start + 1$ under
this numerically convenient scheme where the interpretation
of $start$ versus $end$ is strand independent.

There are a number of implicit conventions in INSDC data that need to be translated into the more explicit FALDO model.
Some of the complicated regions for INSDC are features on a circular chromosome, the most common of which are features that overlap the chromosome's origin of replication.
One such feature is the ``Protein II'' gene from the reverse strand of f1 bacteriophage (ddbj:J02448).
``Protein II'' transcription starts at position 6006 on the reverse strand and ends at position 831 (see Figure~\ref{fig:insdcReverseOverOrigin}).


\begin{figure}
\begin{shaded}
\small
\begin{verbatim}

{"@context" : {"comment"     : "http://www.w3.org/2000/01/rdf-schema#comment",
               "label"       : "http://www.w3.org/2000/01/rdf-schema#label",
               "taxonomy"    : "http://purl.uniprot.org/taxonomy/",
               "ddbj"        : "http://ddbj.nig.ac.jp/ontologies/sequence/",
               "ddbj-record" : "http://ddbj.nig.ac.jp/entry/nucleotide/",
               "ddbj-aa"     : "http://ddbj.nig.ac.jp/entry/protein/",
               "ddbj-seq"    : "http://ddbj.nig.ac.jp/entry/sequence/",
               "identifier"  : "http://purl.org/dc/terms/identifier",
               "faldo"       : "http://biohackathon.org/resource/faldo#",
               "so"          : "http://purl.obolibrary.org/obo/so.owl#"},
 "@id"          : "ddbj-record:J02448",
 "identifier"   : "J02448",
 "comment"      : "Bacteriophage f1, complete genome",
 "ddbj:organism": "taxonomy:10863",
 "ddbj:feature" : {"@id"           : "ddbj-protein:AAA32209.1 ",
                   "@type"         : "so:0000316",
                   "label"         : "Protein II",
                   "faldo:location": {"@type"       : "faldo:Region",
                                       "faldo:begin": {"@type": ["faldo:Position", 
                                                                 "faldo:ExactlyKnownPosition",
                                                                 "faldo:ForwardStrandPosition" ],
                                                       "faldo:position" : "6006",
                                                       "faldo:reference": "ddbj-seq:J02448" },
                                       "faldo:end" : {"@type" : ["faldo:Position", 
                                                                 "faldo:ExactlyKnownPosition",
                                                                 "faldo:ForwardStrandPosition" ],
                                                      "faldo:position"  : "831",
                                                      "faldo:reference" : "ddbj-seq:J02448" }
                                     }
                   }
}
\end{verbatim}
\end{shaded}
\caption{Partial example of using FALDO in JSON-LD\cite{JSONLDFormatSpec} syntax to describe
the CDS ``Protein II'' at \texttt{join(6006..6407,1..831)} on J02448.}
\label{fig:insdcReverseOverOrigin}
\end{figure}


\subsection*{Fuzzy locations}
Feature positions in, for example, INSDC or UniProt, are not always exactly known or described, but we should strive to describe our limited knowledge as accurately as possible.
Take for example the position of the signal peptide annotation shown in Figure~\ref{fig:UniProt}, 
where the protein sequence is known to belong to a family of proteins,
but unfortunately only a part of the amino acid sequence is known. 
The UniProt curator deduced that the signal peptide region only partly overlaps the known sequence fragment.
The same is true in the related INSDC record, were the CDS starts and ends before the known mRNA sequence (see Figure~\ref{fig:DDBJ}).
As demonstrated in the figure, this limited knowledge can be described using the FALDO classes $\mathtt{faldo\colon{}InRangePosition}$ and $\mathtt{faldo\colon{}OneOfPosition}$.

\subsection*{Restriction enzymes}

The task of describing the recognition sites of most restriction
enzymes is quite straightforward, as is describing the cleavage site
of a blunt end cutting enzyme.
However, the cut site of a sticky-end cutting enzyme like HindIII that
leaves an ``overhang'' is more challenging to specify, since it cuts
in a different place on the forward and reverse strands.
Figure~\ref{fig:HindIII} demonstrates how to describe this in FALDO by
specifying start and end positions of the cut site that are on
different strands.

\begin{figure}
\begin{shaded}
\small
\begin{verbatim}
    \/
5'..AAGCTT..3' 
    123456
3'..TTCGAA..5'
        /\
        
:enzymeRestrictionSite a example:EnzymeRestrictionSite ;
       faldo:location :EnzymeRestrictionRegion .
:enzymeRestrictionRegion a faldo:Region ;
       faldo:begin :53_1 ;
       faldo:end   :35_6 .
:53cleaveageSite a example:CleavageSite ;
       faldo:location [a faldo:InBetweenPosition, 
                           SO:0001690 ; # 3' restriction enzyme junction 
                       faldo:after  :53_1 ;        
                       faldo:before :53_2 ] .
:35cleaveageSite a example:CleavageSite ;
       faldo:location [a faldo:InBetweenPosition, 
                           SO:0001689  ; # 5' restriction enzyme junction 
                       faldo:before :35_5 ;
                       #Note the before after reversion due to biological start 
                       faldo:after  :35_6 ]. 
:53_1 a faldo:ExactPosition , faldo:ForwardStrandedPosition ;
      faldo:position 1 .
:53_2 a faldo:ExactPosition , faldo:ForwardStrandedPosition ;
      faldo:position 2 .
:35_5 a faldo:ExactPosition , faldo:ReverseStrandedPosition ;
      faldo:position 5 .
:35_6 a faldo:ExactPosition , faldo:ReverseStrandedPosition ;
      faldo:position 6 .
\end{verbatim}
\end{shaded}
\caption{FALDO representation of HindIII restriction enzyme cleavage site with sticky ends.}
\label{fig:HindIII}
\end{figure}


%\subsection*{More examples}
%\textit{TODO - other ideas:}
%\begin{itemize}
%\item tRNA hairpin (two complementary regions)?
%\item $\beta$-sheet in protein?
%\item RNA-Seq mapping (useful for SAM/BAM)?
%\end{itemize}


